### PR TITLE
Findiff

### PR DESCRIPTION
**Pull request by tnigon**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by tnigon** [comment]
I confirm I am willing to be listed as a maintainer.

File: recipes/findiff/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  noarch: python
+
+requirements:
+  host:
+    - numpy >=1.14.2
+    - python >=3.6
+    - scipy >=1.3.1
+    - sympy >=1.8
```
Comment (dopplershift) [inline recipes/findiff/meta.yaml:22]: I don't think this needs `scipy`, `numpy`, and `sympy` at build time.

**Review by dopplershift — COMMENTED** [review]
Also, please don't touch any other files other than the ones for this recipe, so please undo the commit where you remove the other recipes.

@maroba are you ok being a maintainer for this recipe?

**Comment by dopplershift** [comment]
`pytest-runner` was one that *was* actually required.

**Comment by tnigon** [comment]
Yeah I see that now, just committed changes to add that back in. I also removed @maroba from the maintainer list until we hear otherwise.

**Comment by tnigon** [comment]
Are we okay to proceed with this?

File: recipes/findiff/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - pytest-runner >=5.3.0
+
+test:
+  imports:
+    - findiff
+    - test
```
Comment (chrisburr) [inline recipes/findiff/meta.yaml:32]: A `test` package shouldn't be provided. You'll probably need to patch the sources to fix the setuptools configuration.

File: recipes/findiff/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - pytest-runner >=5.3.0
+
+about:
+  home: "https://github.com/maroba/findiff"
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: "A Python package for finite difference derivatives in any number of dimensions."
+  doc_url:
+  dev_url:
```
Comment (chrisburr) [inline recipes/findiff/meta.yaml:44]: These should either be filled or not included at all
```suggestion
```

File: recipes/findiff/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    - sympy >=1.8
+    - pytest-runner >=5.3.0
+
+test:
+  imports:
+    - findiff
+    - test
+  requires:
+    - pytest >=6.2.3
+    - pytest-runner >=5.3.0
```
Comment (chrisburr) [inline recipes/findiff/meta.yaml:35]: As you're only using an import test these shouldn't be needed.
```suggestion
```

File: recipes/findiff/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    - pip >=21.0.1
+    - pytest-runner >=5.3.0
+  run:
+    - numpy >=1.14.2
+    - python >=3.6
+    - scipy >=1.3.1
+    - sympy >=1.8
+    - pytest-runner >=5.3.0
+
+test:
```
Comment (chrisburr) [inline recipes/findiff/meta.yaml:29]: Can you add a pip check test as well?
```suggestion
test:
      commands:
        - pip check
      requires:
        - pip
```

File: recipes/findiff/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  host:
+    - python >=3.6
+    - pip >=21.0.1
+    - pytest-runner >=5.3.0
+  run:
+    - numpy >=1.14.2
+    - python >=3.6
+    - scipy >=1.3.1
+    - sympy >=1.8
+    - pytest-runner >=5.3.0
```
Comment (dopplershift) [inline recipes/findiff/meta.yaml:27]: I think you only need `pytest-runner` for `build` above, not `run`?

**Comment by chrisburr** [comment]
@tnigon The `tests` package is still being installed, you need to patch [this line](https://github.com/maroba/findiff/blob/e8ca33707e3e25d76bf0f93b2391e466209287b1/setup.py#L55) to replace `find_packages()` with `find_packages(exclude=("tests",))` and make a PR upstream.

This can then either be applied by asking upstream to make a release, or by adding a patch to the sources section: https://conda.io/projects/conda-build/en/latest/resources/define-metadata.html#patches

**Review by dopplershift — APPROVED** [review]

**Comment by maroba** [comment]
> Also, please don't touch any other files other than the ones for this recipe, so please undo the commit where you remove the other recipes.
> 
> @maroba are you ok being a maintainer for this recipe?

Yes, that is ok for me.

**Pull request merged**